libs/core/langchain_core/runnables/base.py

```diff
@@ -2006,6 +2007,19 @@ async def _atransform_stream_with_config(
             if hasattr(iterator_, "aclose"):
                 await iterator_.aclose()
 
+    def as_tool(
+        self,
+        *,
+        name: Optional[str] = None,
+        description: Optional[str] = None,
+        arg_types: Optional[Dict[str, Type]] = None,
+    ) -> BaseTool:
+        from langchain_core.tools import convert_runnable_to_tool
```

Comment:
nit: comment that this is for import time optimization i assume

Reply:
I import here to avoid a circular import